test/beast/zlib/inflate_stream.cpp

```diff
@@ -360,6 +380,143 @@ class inflate_stream_test : public beast::unit_test::suite
+               0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+               0x00, 0x00, 0x00, 0x00, 0x00, 0x06}, {});
+    }
+
+    std::string check(
+        std::initializer_list<std::uint8_t> const& in,
+        error_code expected,
+        std::size_t window_size = 15,
+        std::size_t len = -1)
+    {
+        std::string out(1024, 0);
+        z_params zs;
+        inflate_stream is;
+        is.reset(static_cast<int>(window_size));
+        boost::system::error_code ec;
+
+        zs.next_in = &*in.begin();
+        zs.next_out = &out[0];
+        zs.avail_in = std::min(in.size(), len);
+        zs.avail_out = out.size();
+
+        while (zs.avail_in > 0 && !ec)
+        {
+            is.write(zs, Flush::sync, ec);
+            auto n = std::min(zs.avail_in, len);
+            zs.next_in = static_cast<char const*>(zs.next_in) + n;
+            zs.avail_in -= n;
+        }
+
+        BEAST_EXPECT(ec == expected);
+        return out;
+    }
+
+    void testInflateErrors()
+    {
+        check({0x00, 0x00, 0x00, 0x00, 0x00},
+            error::invalid_stored_length);
+        check({0x03, 0x00},
+            error::end_of_stream);
+        check({0x06},
+            error::invalid_block_type);
+        check({0xfc, 0x00, 0x00},
+            error::too_many_symbols);
+        check({0x04, 0x00, 0xfe, 0xff},
+            error::incomplete_length_set);
+        check({0x04, 0x00, 0x24, 0x49, 0x00},
+            error::invalid_bit_length_repeat);
+        check({0x04, 0x00, 0x24, 0xe9, 0xff, 0xff},
+            error::invalid_bit_length_repeat);
+        check({0x04, 0x00, 0x24, 0xe9, 0xff, 0x6d},
+            error::missing_eob);
+        check({0x04, 0x80, 0x49, 0x92, 0x24, 0x49, 0x92, 0x24,
+               0x71, 0xff, 0xff, 0x93, 0x11, 0x00},
+            error::over_subscribed_length);
+        check({0x04, 0x80, 0x49, 0x92, 0x24, 0x0f, 0xb4, 0xff,
+               0xff, 0xc3, 0x84},
+            error::incomplete_length_set);
+        check({0x04, 0xc0, 0x81, 0x08, 0x00, 0x00, 0x00, 0x00,
+               0x20, 0x7f, 0xeb, 0x0b, 0x00, 0x00},
+            error::invalid_literal_length);
+        check({0x02, 0x7e, 0xff, 0xff},
+            error::invalid_distance_code);
+        check({0x0c, 0xc0, 0x81, 0x00, 0x00, 0x00, 0x00, 0x00,
+               0x90, 0xff, 0x6b, 0x04, 0x00},
+            error::invalid_distance);
+        check({0x05,0xe0, 0x81, 0x91, 0x24, 0xcb, 0xb2, 0x2c,
+               0x49, 0xe2, 0x0f, 0x2e, 0x8b, 0x9a, 0x47, 0x56,
+               0x9f, 0xfb, 0xfe, 0xec, 0xd2, 0xff, 0x1f},
+            error::end_of_stream);
+        check({0xed, 0xc0, 0x01, 0x01, 0x00, 0x00, 0x00, 0x40,
+               0x20, 0xff, 0x57, 0x1b, 0x42, 0x2c, 0x4f},
+            error::end_of_stream);
+        check({0x02, 0x08, 0x20, 0x80, 0x00, 0x03, 0x00},
+            error::end_of_stream);
+        // TODO: Excess data (from golang test inflate suite), should this be an error?
```

Comment:
IMO excess data in inflate is definitely an error; the user should be able to provide the right compressed length